index.src.html

```diff
@@ -2355,13 +2373,49 @@
             |clone|.
     2. Return |clone|.
 
+: To check if a {{AudioDataInit}} is a
+    <dfn>valid AudioDataInit</dfn>, run these steps:
+:: 1. If {{AudioDataInit/sampleRate}} = `0`, return `false`.
+    2. If {{AudioDataInit/numberOfFrames}} = `0`, return `false`.
+    3. If {{AudioDataInit/numberOfChannels}} = `0`, return `false`.
+    4. Verify {{AudioDataInit/data}} has enough data by running the following
+        steps:
+        1. Let |totalSamples| be the product of multiplying
+            {{AudioDataInit/numberOfFrames}} by
+            {{AudioDataInit/numberOfChannels}}.
+        2. Let |bytesPerSample| be the number of bytes per sample, as defined by
+            the {{AudioDataInit/format}}.
+        3. Let |totalSize| be the product of multiplying |bytesPerSample| with
+            |totalSamples|.
+        4. Let |dataSize| be the size in bytes of {{AudioDataInit/data}}.
+        5. If |dataSize| is less than |totalSize|, return false.
+    5. Return `true`.
+
+<div class='note'>
+Note: An {{AudioDataInit}}'s {{AudioDataInit/data}} must have the following
+    memory layout:
+    - If {{AudioDataInit/format}} describes an [=interleaved=]
+        {{AudioSampleFormat}}, {{AudioDataInit/data}} should contain
+        one block of samples for each of the {{AudioDataInit/numberOfFrames}}
+        frames, layed out end to end, as a single plane. Within each block of
+        samples, the `nth` sample corresponds to the `nth` of the
+        {{AudioDataInit/numberOfChannels}} channels.
+    - If {{AudioDataInit/format}} describes a [=planar=] {{AudioSampleFormat}},
+        each channel's samples must be consecutively layed out in memory,
+        forming a plane. Every plane must be layed out directly after the
+        previous plane, preserving the ordering of channels. This means that the
```

Comment:
This is not observable, is it? Best not to talk about it, because it's problematic when doing channel reordering.